tools/functional-tester/etcd-tester/tester.go

```diff
@@ -83,6 +83,16 @@ func (tt *tester) runLoop() {
 			plog.Printf("[round#%d case#%d] recovered failure", i, j)
 
 			if !tt.consistencyCheck {
+				plog.Printf("[round#%d case#%d] canceling the stressers...", i, j)
+				for _, s := range tt.cluster.Stressers {
+					s.Cancel()
+				}
+				plog.Printf("[round#%d case#%d] canceled stressers", i, j)
+				time.Sleep(3 * time.Second)
```

Comment:
So previous code has no Sleep here (not waiting for in-flight stressers to finish). Now it waits for them to exit. And it's been running without failure.

![screenshot from 2016-04-26 14-38-31](https://cloud.githubusercontent.com/assets/6799218/14835036/93f9118e-0bbc-11e6-9bb4-35d52ac2b941.png)